packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx

```diff
@@ -46,14 +46,19 @@ export class AtomicSearchBox implements AtomicSearchBoxOptions {
     'atomic-search-box-'
   );
 
-  private context!: InterfaceContext;
+  public context!: InterfaceContext;
   private searchBox!: SearchBox;
   private unsubscribe: Unsubscribe = () => {};
   private inputRef!: HTMLInputElement;
   private valuesRef!: HTMLElement;
   private containerRef!: HTMLElement;
   private combobox!: Combobox;
 
+  @State() strings = {
```

Comment:
As discussed this morning, it would be awesome if we had something to abstract the concepts of localized string + how to handle the `updateLocaleStrings`.

A decorator, like you proposed would be nice.

So that a newcomer can simply copy paste the recipe to add new localized string, or to create a new component with localized string, and it's super straightforward to do.

Reply:
Hmm that's what I do to the extent I can in the Initialization decorator, it calls the `updateLocaleStrings` directly, only the user has to do is define the update method to update the state.

To have the right types, we need to define a state on line 57, but we can't access the context property yet. I can look a bit further

Reply:
@olamothe 
Okay I now added methods directly in the state, the state is then automatically updated when the language state (in the Initialization decorator of the initialization-utils.tsx file, see latest commit)